jina/types/document/__init__.py

```diff
@@ -534,3 +534,24 @@ def MergeFrom(self, doc: 'Document'):
 
     def CopyFrom(self, doc: 'Document'):
         self._document.CopyFrom(doc.as_pb_object)
+
+    def traverse_apply(self, traversal_paths: Tuple[str], apply_func: Callable, *args, **kwargs) -> None:
```

Comment:
Not sure, but `traverse_apply` feels like it may belong into `DocumentSet` class maybe?

Reply:
I have a better idea, the document handle `traverse` and document set handles traverse_all, I'll update the code, one sec.

Reply:
Seems like a good idea! But anyway good to iterate on this, and to break things to learn the insights of how the feature works